app/controllers/dor_controller.rb

```diff
@@ -51,11 +51,15 @@ def reindex
     obj = Dor.load_instance params[:pid]
     @solr_doc = obj.to_solr
     Dor::SearchService.solr.add(@solr_doc, :add_attributes => {:commitWithin => 1000}) unless obj.nil?
-    index_logger.info "updated index for #{params[:pid]}"
+    message = "Successfully updated index for #{params[:pid]}"
+    index_logger.info message
+    flash[:notice] = message
+    redirect_to :back
   rescue ActiveFedora::ObjectNotFoundError # => e
-    index_logger.info "failed to update index for #{params[:pid]}, object not found in Fedora"
-    render :status => 500, :text => 'Object doesnt exist in Fedora.'
-    return
+    message = "Failed to update index for #{params[:pid]}, object not found in Fedora"
+    index_logger.info message
```

Comment:
this has been refactored in the profiling indexing work; i agreed with @drh-stanford's suggestion, apparently:
```ruby
# DorController.reindex
  def reindex
    @solr_doc = Argo::Indexer.reindex_pid params[:pid], index_logger
    Dor::SearchService.solr.commit # reindex_pid doesn't commit
  rescue ActiveFedora::ObjectNotFoundError # => e
    render :status => 404, :text => 'Object does not exist in Fedora.'
    return
  end
```

```ruby
Argo::Indexer.reindex_pid
    def self.reindex_pid(pid, index_logger = nil, should_raise_errors = true)
      index_logger ||= default_index_logger
      obj = Dor.load_instance pid
      solr_doc = reindex_object obj
      index_logger.info "updated index for #{pid}"
      solr_doc
    rescue StandardError => se
      if se.is_a? ActiveFedora::ObjectNotFoundError
        index_logger.warn "failed to update index for #{pid}, object not found in Fedora"
      else
        index_logger.warn "failed to update index for #{pid}, unexpected StandardError, see main app log: #{se.backtrace}"
      end
      raise se if should_raise_errors
    rescue Exception => ex
      index_logger.error "failed to update index for #{pid}, unexpected Exception, see main app log: #{ex.backtrace}"
      raise ex # don't swallow anything worse than StandardError
    end
```